### PR TITLE
Support `require_data_stream` new config option

### LINKED ISSUE #110: Add config to add `require_data_stream` to indexing requests
https://github.com/elastic/elasticsearch/pull/101872 has been merged and will be part of 8.13.0. Let's update go-docappender to optionally set this flag in indexing requests.

### PR DESCRIPTION
**Pull request by marclop**
Adds a new `RequireDataStream` option to the Appender, and BulkIndexer configurations. When set, it will send `?require_data_stream=true` in the `_bulk` request to Elasticsearch.

Closes #110

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -90,6 +90,14 @@ type Config struct {
 	// If Pipeline is empty, no ingest pipeline will be specified in the Bulk request.
 	Pipeline string
 
+	// RequireDataStream forces the auto creation of the underlying index.
```
Comment (carsonip) [inline config.go:93]: This line doesn't read very right, although the explanation below is correct.

Comment (marclop) [reply config.go]: you're right, just re-formatted it.

**Review by endorama — DISMISSED** [review]
Looks good to me!

File: config.go [hunk 2]
```diff
@@ -90,6 +90,13 @@ type Config struct {
 	// If Pipeline is empty, no ingest pipeline will be specified in the Bulk request.
 	Pipeline string
 
+	// RequireDataStream, If set to true, an index will be created only if a
```
Comment (endorama) [inline config.go:93]: A true nitpick but could be beneficial to explain the default behavior and why one may want enable this option, to help readers unfamiliar with it.

Comment (marclop) [reply config.go:93]: Documented the default behavior commit below.

**Review by carsonip — APPROVED** [review]

**Pull request merged**